ethcore/src/miner/miner.rs

```diff
@@ -1114,77 +1115,26 @@ impl miner::MinerService for Miner {
 					max_len,
 					ordering,
 				},
-			)
+			);
+
+			pending.retain(|tx| {
```

Comment:
this is the `real fix`

Reply:
Would have been hard to spot otherwise! :D

Reply:
just curious, why not tack on the `retain` call to the call to `pending()`?

Reply:
See [my earlier comment](https://github.com/paritytech/parity-ethereum/pull/11227/files#r343253293)